### ISSUE TITLE
Evaluation of 'Tartan-T.ckpt' on Sintel training set

### ISSUE THREAD

**Issue by RIA1159** [msu-video-group/memfof#8]
Hi there！
I notice that the Tab3 only contains the evaluation results on Sintel and KITTI-15 test sets. For ease of comparison, I’d like to know how the model performs on the training set.
So I change the "val_datasets" and "restore_ckpt" in ./config/eval/sintel.json to ["sintel-clean", "sintel-final"] and "./ckpts/Tartan-T.ckpt", and run the evaluation with: python3 -m scripts.evaluate --cfg config/eval/sintel.json

The evaluation results are as follows:
val-sintel-clean-epe    │    0.8569297194480896
 val-sintel-final-epe    │    5.8559417724609375

Compared to VideoFlow(3-frame, training data T): 
sintel-clean: 1.03
sintel-final: 2.19

There has been a significant improvement on sintel-clean, but the error on sintel-final is unusually high. I’d like to know if this result is normal, or if there are any other configurations I should change.
Looking forward for your replay! :)

**Comment by egorchistov** [comment]
Hi!

Sorry for the late reply. I think our subset naming convention may have caused some confusion. When you specified `"val_datasets": ["sintel-clean", "sintel-final"]`, the metrics were computed only on the validation subset we selected — specifically alley_2, ambush_6, bamboo_2, shaman_3, and temple_3.

As I understand it from the VideoFlow paper, their reported numbers (1.03 for Sintel train clean and 2.19 for Sintel train final) are computed on the full training set (please correct me if I’m mistaken).

To compute metrics on the full training set, as done in VideoFlow, please replace all `train` labels with `val` labels in the following file: [config/splits/sintel.json](https://github.com/msu-video-group/memfof/blob/dev/config/splits/sintel.json).

Could you recompute the metrics and let me know if the results look more reasonable?

Thanks!

**Comment by RIA1159** [comment]
Thanks for your reply!
I've changed the 'train' labels to 'val' in both  ./config/splits/sintel.json and ./config/splits/kitti.json  and recomputed the metrics with the "./ckpts/Tartan-T.ckpt",  however the results seems even worse:

val-sintel-clean-epe    │     1.487846851348877
val-sintel-final-epe    │     6.905431270599365
    val-kitti-epe       │     3.6054182052612305     
    val-kitti-f1        │     12.11718463897705

However, after I retrained the model locally using the Tartan.ckpt weights with the command python3 -m scripts.train --cfg config/train/Tartan-T.json, I obtained the new weights 'Tartan-T_step=120000.ckpt'. Its metrics are as follows:

val-sintel-clean-epe    │    1.152282953262329
val-sintel-final-epe    │    3.0748190879821777
    val-kitti-epe       │    2.9475104808807373     
    val-kitti-f1        │    10.047956466674805 

This result seems more reasonable, as it is close to the values in Table 10 of the supplementary materials of the paper. I’m wondering if the released weight './ckpts/Tartan-T.ckpt' might be different from the one in the table.

**Comment by egorchistov** [comment]
The published weights should be the same as those we used for fine-tuning, so the results are expected to match. However, the numbers you obtained look strange.

I will double-check this and get back to you. It may take some time.

Thanks for pointing this out!

**Comment by RIA1159** [comment]
I found the problem, I used the different padding settings.
